### PR TITLE
type hints

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

- Adds type hints now that we have dropped PHP 7.4
- Corrects docstring type hints to match
- Removes the API key check, now that there are type hints, the lib will throw an error at runtime if required params (like the API key) are not passed
- Corrected all magic method params to conform to what PHP expects

# Testing

Tests caught various items needed fixing. In PHP, if you don't pass params that match the definitions you'll get runtime errors. This will help us develop but also will help users know what to pass in. Re-ran all tests every few files I edited.

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Review by nwithan8 — APPROVED** [review]
Ah `mixed`, the `Any` of PHP typing, lol.

**Comment by Justintime50** [comment]
> Ah `mixed`, the `Any` of PHP typing, lol.

Haha yeah... a lot of that can go away when we deserialize to explicit objects, unfortunately at present everything is just a basic object or array and needs mixed

File: lib/EasyPost/EasyPostClient.php [hunk 1]
```diff
@@ -96,22 +95,16 @@ public function __construct(
         $this->requestEvent = new RequestHook();
         $this->responseEvent = new ResponseHook();
         $this->httpClient = new Client();
-
-        if (!$this->apiKey) {
-            throw new MissingParameterException(
-                'No API key provided. See https://www.easypost.com/docs for details, or contact ' . Constants::SUPPORT_EMAIL . ' for assistance.' // phpcs:ignore
-            );
-        }
```
Comment (jchen293) [inline lib/EasyPost/EasyPostClient.php:104]: why was this removed?

**Review by jchen293 — APPROVED** [review]

File: lib/EasyPost/EasyPostClient.php (hunk 1 above)
Comment (Justintime50) [reply lib/EasyPost/EasyPostClient.php:104]: As mentioned in the summary: `Removes the API key check, now that there are type hints, the lib will throw an error at runtime if required params (like the API key) are not passed`
